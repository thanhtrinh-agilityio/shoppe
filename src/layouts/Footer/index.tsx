import { Box, Flex, Text } from '@chakra-ui/react'

// Components
import Icon from '@components/Icon'
import NavBar from '@components/NavBar'

// Constants
import { NAV_LIST, SOCIAL_ICONS } from '@constants/index'

const Footer = () => {
  return (
    <Box
      as="footer"
      pt="37px"
      pb="108px"
      borderTop="1px"
      borderTopColor="lightSilver"
      maxWidth="1248px"
      margin="0 auto"
    >
      <Flex justifyContent="space-between">
        <NavBar paddingTop="10px" navList={NAV_LIST} />
        <Flex
          borderBottom="1px"
          borderBottomColor="dark"
          justifyContent="space-between"
          alignItems="center"
        >
          <Text fontSize="base" color="secondary" marginBottom="10px">
            Give an email, get the newsletter.
          </Text>
          <Icon
            alt="Right arrow"
            ml="128px"
            srcIcon="/images/iconRight.svg"
            width="25px"
            height="9px"
          />
        </Flex>
      </Flex>
      <Flex mt="48px" justifyContent="space-between" alignItems="center">
        <Text>
          <Text as="b" variant="primary">
            Â© 2021 Shelly.
          </Text>{' '}
          Terms of use
          <Text as="b" variant="primary">
            {' '}
            and
          </Text>{' '}
          privacy policy
        </Text>
        <Flex w="150px" justifyContent="space-between">
          {SOCIAL_ICONS.map((icon) => (
            <Icon
              width="18px"
              height="18px"
              key={icon.id}
              alt={icon.alt}
              srcIcon={icon.iconURL}
            />
          ))}
        </Flex>
      </Flex>
    </Box>
  )
}

export default Footer
