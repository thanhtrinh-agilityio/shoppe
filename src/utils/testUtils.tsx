import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { CartProvider } from '@contexts/CartProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <CartProvider>{component}</CartProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
