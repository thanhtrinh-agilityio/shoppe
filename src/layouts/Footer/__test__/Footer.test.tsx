// Utils
import { render } from '@utils/testUtils'

// Components
import Footer from '@layouts/Footer'

describe('Footer render', () => {
  it('Should Footer match Snapshot', () => {
    const { container } = render(<Footer />)

    expect(container).toMatchSnapshot()
  })
})
