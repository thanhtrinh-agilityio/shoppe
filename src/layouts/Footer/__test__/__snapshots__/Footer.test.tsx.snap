// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render Should Footer match Snapshot 1`] = `
<div>
  <footer
    class="css-xmbrgk"
  >
    <div
      class="css-gg4vpm"
    >
      <nav
        class="css-fd0xex"
      >
        <a
          href="/contact"
        >
          Contact
        </a>
        <a
          href="/terms-of-services"
        >
          Terms of services
        </a>
        <a
          href="/shipping-and-return"
        >
          Shipping and Return
        </a>
      </nav>
      <div
        class="css-11xbhwq"
      >
        <p
          class="chakra-text css-ozfm6u"
        >
          Give an email, get the newsletter.
        </p>
        <div
          class="css-1o1su5v"
        >
          <img
            alt="Right arrow"
            data-nimg="fill"
            data-testid="image"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/images/iconRight.svg?w=3840&q=75"
            srcset="/images/iconRight.svg?w=640&q=75 640w, /images/iconRight.svg?w=750&q=75 750w, /images/iconRight.svg?w=828&q=75 828w, /images/iconRight.svg?w=1080&q=75 1080w, /images/iconRight.svg?w=1200&q=75 1200w, /images/iconRight.svg?w=1920&q=75 1920w, /images/iconRight.svg?w=2048&q=75 2048w, /images/iconRight.svg?w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
    </div>
    <div
      class="css-167212l"
    >
      <p
        class="chakra-text css-tujbob"
      >
        <b
          class="chakra-text css-lz4kvn"
        >
          Â© 2021 Shelly.
        </b>
         
        Terms of use
        <b
          class="chakra-text css-lz4kvn"
        >
           
          and
        </b>
         
        privacy policy
      </p>
      <div
        class="css-st3z69"
      >
        <div
          class="css-14b8luf"
        >
          <img
            alt="LinkIn Icon"
            data-nimg="fill"
            data-testid="image"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/images/inIcon.svg?w=3840&q=75"
            srcset="/images/inIcon.svg?w=640&q=75 640w, /images/inIcon.svg?w=750&q=75 750w, /images/inIcon.svg?w=828&q=75 828w, /images/inIcon.svg?w=1080&q=75 1080w, /images/inIcon.svg?w=1200&q=75 1200w, /images/inIcon.svg?w=1920&q=75 1920w, /images/inIcon.svg?w=2048&q=75 2048w, /images/inIcon.svg?w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="css-14b8luf"
        >
          <img
            alt="Facebook icon"
            data-nimg="fill"
            data-testid="image"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/images/facebook.svg?w=3840&q=75"
            srcset="/images/facebook.svg?w=640&q=75 640w, /images/facebook.svg?w=750&q=75 750w, /images/facebook.svg?w=828&q=75 828w, /images/facebook.svg?w=1080&q=75 1080w, /images/facebook.svg?w=1200&q=75 1200w, /images/facebook.svg?w=1920&q=75 1920w, /images/facebook.svg?w=2048&q=75 2048w, /images/facebook.svg?w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="css-14b8luf"
        >
          <img
            alt="Instagram icon"
            data-nimg="fill"
            data-testid="image"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/images/instagram.svg?w=3840&q=75"
            srcset="/images/instagram.svg?w=640&q=75 640w, /images/instagram.svg?w=750&q=75 750w, /images/instagram.svg?w=828&q=75 828w, /images/instagram.svg?w=1080&q=75 1080w, /images/instagram.svg?w=1200&q=75 1200w, /images/instagram.svg?w=1920&q=75 1920w, /images/instagram.svg?w=2048&q=75 2048w, /images/instagram.svg?w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="css-14b8luf"
        >
          <img
            alt="Twitter icon"
            data-nimg="fill"
            data-testid="image"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/images/twitter.svg?w=3840&q=75"
            srcset="/images/twitter.svg?w=640&q=75 640w, /images/twitter.svg?w=750&q=75 750w, /images/twitter.svg?w=828&q=75 828w, /images/twitter.svg?w=1080&q=75 1080w, /images/twitter.svg?w=1200&q=75 1200w, /images/twitter.svg?w=1920&q=75 1920w, /images/twitter.svg?w=2048&q=75 2048w, /images/twitter.svg?w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
      </div>
    </div>
  </footer>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
